lib/factories/in-memory-delay-interceptor.factory.ts

```diff
@@ -0,0 +1,8 @@
+import { InMemoryDBConfig } from '../interfaces';
+import { SimulatedDelayInterceptor } from '../interceptors/simulated-delay.interceptor';
+
+export function inMemoryDBDelayInterceptorFactory(
+    featureConfig: Partial<InMemoryDBConfig>
+) {
+    return () => new SimulatedDelayInterceptor(featureConfig);
```

Comment:
Nit, should `SimulatedDelayInterceptor` be aware of `InMemoryDBConfig`?
We could introduce a  `NoopInterceptor` which does nothing which saves us a few function calls:

- if `delay` is provided, use `SimulatedDelayInterceptor(config.delay)`
- if `delay` is not provided, use `NoopInterceptor`

Reply:
I like this idea @timdeschryver .